bounter/count_min_sketch.py

```diff
@@ -62,17 +63,17 @@ def __init__(self, size_mb=64, width=None, depth=None, algorithm='conservative')
                 If width is not provided, the algorithm chooses the maximum width to fill the available size.
                 The more, the better, should be very large, preferably in the same order of magnitude as the cardinality
                 of the counted set.
-            algorithm (str): controls the algorithm used for counting
-                - 'conservative' (default)
-                - 'log1024'
-                - 'log8'
+            log_counting (int): Use logarithmic approximate counter value for reduced bucket size:
+                - None (default): 4B, no counter error
+                - 1024: 2B, value approximation error ~2% for values larger than 2048
+                - 8: 1B, value approximation error ~30% for values larger than 16
         """
 
-        cell_size = CountMinSketch.cell_size(algorithm);
+        cell_size = CountMinSketch.cell_size(log_counting)
         self.cell_size_v = cell_size
 
         if not isinstance(size_mb, int):
-            raise ValueError("size_mb must be an integer!")
+            raise ValueError("size_mb must be an integer.")
```

Comment:
And it is what...? Be more concrete in the error messages (here and elsewhere).

Tantalizingly vague error messages are some of the most frustrating things for users :)

Reply:
In which way should I be more concrete?
Can you be more specific? :D

Reply:
Sure -- `"size_mb must be an integer, not %r" % size_mb` .

Similarly in the other messages -- when we're complaining X is not Y (bad value), spit back both X and Y.